sycl/doc/extensions/proposed/sycl_ext_oneapi_usm_malloc_properties.asciidoc

```diff
@@ -0,0 +1,1615 @@
+Runtime properties in `propList` will not appear as properties of the returned `annotated_ptr`.
+The returned `annotated_ptr` will have the property `usm_kind<sycl::usm::alloc::device>`.
+
+Throws a synchronous `exception` with the
+`errc::feature_not_supported` error code if the device
+does not have `aspect::usm_device_allocations`.
+
+An error is reported if `propList` contains
+a `usm_kind` property with value different than `sycl::usm::alloc::device`.
+
+Available only if `propertyListA` and `propertyListB` are specializations of the `properties` class.
+
+a@
+[source]
+----
+template <typename T,
+          typename propertyListA,
+          typename propertyListB>
+annotated_ptr<T, propertyListB>
+annotated_malloc_device(
+   size_t count,
+   const queue& syclQueue,
+   const propertyListA &propList = properties{})
+----
+a@ Simplified form where `syclQueue` provides the `device`
+and `context`.
+
+Zero or more runtime and compile-time constant properties can be provided to the allocation function
+via an instance of `properties`.
+The compile-time constant properties in `propList` will appear as properties of the returned `annotated_ptr`.
+Runtime properties in `propList` will not appear as properties of the returned `annotated_ptr`.
+The returned `annotated_ptr` will have the property `usm_kind<sycl::usm::alloc::device>`.
+
+Throws a synchronous `exception` with the
+`errc::feature_not_supported` error code if the device
+does not have `aspect::usm_device_allocations`.
+
+An error is reported if `propList` contains
+a `usm_kind` property with value different than `sycl::usm::alloc::device`.
+
+Available only if `propertyListA` and `propertyListB` are specializations of the `properties` class.
+
+a@
+[source]
+----
+template<typename propertyListA, typename propertyListB>
+annotated_ptr<void, propertyListB>
+annotated_aligned_alloc_device(
+   size_t alignment,
+   size_t numBytes,
+   const device& syclDevice,
+   const context& syclContext,
+   const propertyListA &propList = properties{})
+----
+a@ Returns an `annotated_ptr` containing a raw pointer to the newly allocated memory on
+the specified `device` on success.
+The memory will have `alignment`-byte alignment.
+The allocation size is specified in bytes.  This memory is not accessible on
+the host.  Memory allocated by `annotated_aligned_alloc_device` must be
+deallocated with `sycl::free` to avoid memory leaks.
+On failure, the raw pointer of the returned `annotated_ptr` will be `nullptr`.
+
+Devices may only permit certain alignments. 
+
+Zero or more runtime and compile-time constant properties can be provided to the allocation function
+via an instance of `properties`.
+The compile-time constant properties in `propList` will appear as properties of the returned `annotated_ptr`.
+Runtime properties in `propList` will not appear as properties of the returned `annotated_ptr`.
+The returned `annotated_ptr` will have the property `usm_kind<sycl::usm::alloc::device>`.
```

Comment:
The returned `annotated_ptr` won't have the alignment property because the alignment is specified at runtime (via the parameter), not at compile-time.

Reply:
Right, makes sense.  See my comment above about noting this in the examples section, though.

Reply:
I updated the examples to note when alignment does appear on the returned annotated_ptr and when it does not.